drivers/usb/device/usb_dc_stm32.c

```diff
@@ -36,6 +37,23 @@ LOG_MODULE_REGISTER(usb_dc_stm32);
 #error "Only one interface should be enabled at a time, OTG FS or OTG HS"
 #endif
 
+/*
+ * Some STM32U5xx parts support a USB HS PHY which is clocked by the HSE clock or PLL1_P
+ * clock directly.  This requires specific configuration.
+ */
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_otghs) && defined(CONFIG_SOC_SERIES_STM32U5X)
```

Comment:
I think this condition should be replaced with `DT_HAS_COMPAT_STATUS_OKAY(st_stm32u5_otghs_phy)`.
Apply in the whole file.